src/DotLiquid.Tests/ConditionTests.cs

```diff
@@ -695,6 +699,80 @@ public void TestEqualOperatorsWorksOnEnum()
             AssertEvaluatesFalse("enum", "!=", "'Yes'");
         }
 
+        [Test]
+        public void TestBlankObject()
+        {
+            _context = new Context(CultureInfo.InvariantCulture);
+            _context["dict"] = new Dictionary<string, string> { { "abc", "xyz" } };
+            _context["empty_dict"] = new Dictionary<string, string> { };
+            _context["list"] = new List<string> { "abc" };
+            _context["empty_list"] = new List<string> { };
+            _context["arr"] = new string[] { "foo" };
+            _context["empty_arr"] = new string[] { };
+            _context["a_drop"] = new DummyDrop();
+
+            // self check
+            AssertEvaluatesFalse(left: "blank", op: "==", right: "blank");

```

Comment:
Why is the blank string not equal to itself?

Reply:
Added to my ruby tests - blank is not blank and empty is not empty.

https://replit.com/@microalps/LiquidBlankTests#main.rb

Reply:
Also if you're looking for documentation about this, it simply doesn't exist. The entire test ruby script fails miserably without the line `require 'active_support/all'` because ruby and liquid doesn't define blank or empty. These are left to individual classes to define - and in this case Rails extends string and other common classes to define these methods for you. You'll find some more explanation on [stackoverflow](https://stackoverflow.com/questions/11029256/difference-between-nil-blank-and-empty)

Reply:
🤯

Reply:
As the blank concept is inherited from Ruby on Rails, it appears logical to mimic that source's behavior.

Reply:
Consider capturing some explanation as an inline code comment for future maintainers.